robot/pmm/resources/pmm.py

```diff
@@ -478,3 +478,121 @@ def select_app_from_all_items(self, app_name):
+        self.selenium.wait_until_location_contains(
+            "/lightning/setup/CustomMetadata/home"
+        )
+
+    def select_frame_with_value(self, value):
+        """Selects frame identified by the given value
+        value should be the 'id', 'title' or 'name' attribute value of the web element used to identify the frame
+        """
+        locator = pmm_lex_locators["frame"]
+        locator = self.format_all(locator, value)
+        self.selenium.select_frame(locator)
+
+    def manage_records_bucketed_value(self):
+        """click on Manage Records for Bucketed Value"""
+        self.select_frame_with_value(
+            "All Custom Metadata Types ~ Salesforce - Developer Edition"
+        )
+        locator = pmm_lex_locators["custom_metadata_link"]
+        self.selenium.wait_until_page_contains_element(locator)
+        self.selenium.click_element(locator)
+
+    def click_custom_metadata_button(self, button):
+        """Clicks on a button within the custom metadata iFrame"""
+        if button == "New":
+            locator = pmm_lex_locators["custom_metadata_button"].format(button)
+            self.select_frame_with_value(
+                "Bucketed Values ~ Salesforce - Developer Edition"
+            )
+            self.selenium.wait_until_page_contains_element(locator)
+            self.selenium.click_element(locator)
+        else:
+            locator = pmm_lex_locators["custom_metadata_button_title"].format(button)
+            self.selenium.wait_until_page_contains_element(locator)
+            self.selenium.click_element(locator)
+
+    def populate_bucketed_values(self, **kwargs):
+        """Populates text field and lookup values within the custom metadata iFrame"""
+        for key, value in kwargs.items():
+            if key == "Label":
+                self.select_frame_with_value(
+                    "Bucketed Value ~ Salesforce - Developer Edition"
+                )
+                locator_field = pmm_lex_locators["custom_metadata_field"].format(key)
+                if self.check_if_element_exists(locator_field):
+                    self.selenium.get_webelement(locator_field).click()
+                    self.selenium.set_focus_to_element(locator_field)
+                    self.selenium.clear_element_text(locator_field)
+                    self.selenium.get_webelement(locator_field).send_keys(value)
+                else:
+                    self.builtin.log(f"Element {key} not found")
+            elif key == "Bucket":
+                print("test")
+                locator_value = pmm_lex_locators["custom_metadata_lookup_field"].format(
+                    key
+                )
+                if self.check_if_element_exists(locator_value):
+                    self.selenium.get_webelement(locator_value).click()
+                    self.selenium.set_focus_to_element(locator_value)
+                    self.selenium.clear_element_text(locator_value)
```

Comment:
Since these steps are repetitive in each if/else block you might want to just wrap them in a single function.

Reply:
Fixed